### PR TITLE
Adding a check conflict function to create potential maps

### PR DESCRIPTION
**Pull request by qchuchu**
# Changes
Adding a check conflict function, that takes as a parameter a cell and a player_index, and checks if there are potential conflicts on the hypothetical cell, and solve them.

# Testing
check_conflict(array([0, 3, 1], 1) should return array([0, 3, 1])

### CODE REVIEW

**Review by Barth2501 — APPROVED** [review]
seems fine to me

File: lib/alpha-beta.py [hunk 1]
```diff
@@ -64,11 +65,52 @@ def get_successors(state: ndarray, species: int):
...
+            else:
+                probability_of_win = nb_player / (2 * nb_opponent)
+            battle_won = binomial(1, probability_of_win)
+            if battle_won:
+                cell[opponent_index] = 0
+                cell[player_index] = binomial(nb_player, probability_of_win)
+            else:
+                cell[player_index] = 0
+                cell[opponent_index] = binomial(nb_opponent, 1 - probability_of_win)
+    return cell
```
Comment (hinosxz) [inline lib/alpha-beta.py:112]: Can we avoid updating the parameter `cell` and create a copy of it?

**Review by hinosxz — CHANGES_REQUESTED** [review]
Looks great, only one blocker 👏

File: lib/alpha-beta.py [hunk 2]
```diff
@@ -124,3 +167,7 @@ def min_value(s: ndarray, alpha: int, beta: int, depth: int):
 #         [[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0], [1, 0, 0], [0, 0, 0]],
 #     ])
 # print(alphabeta_search("vampire", example))
+
+if __name__ == "__main__":
```
Comment (hinosxz) [inline lib/alpha-beta.py:171]: Might not be useful 🙄

**Review by hinosxz — APPROVED** [review]
Nice work, looks great 👍

**Pull request merged**